src/helpers/analyticsEvents/studyGuides/openCloseColorKey.ts

```diff
@@ -0,0 +1,24 @@
+import { createSelector } from 'reselect';
+import * as selectNavigation from '../../../app/navigation/selectors';
+import { AnalyticsEvent } from '../event';
+
+const openText = 'REX Study guides (open color key)';
+const closeText = 'REX Study guides (close color key)';
+
+export const selector = createSelector(
+  selectNavigation.pathname,
+  (pathname) => ({pathname})
+);
+
+export const track = (
+  {pathname}: ReturnType<typeof selector>,
+  open?: boolean
+): AnalyticsEvent | void => {
+  return {
+    getGoogleAnalyticsPayload: () => ({
+      eventAction: 'button',
+      eventCategory: open ? closeText : openText, // async call
```

Comment:
@janKozie1

Reply:
I'm not really sure if the comment makes things clearer. I feel like the ternary should be like it was before, just change how the event is dispatched in the component. 
I think the best would be to just add an `useEffect` that fires the event when the `open` changes, but that's just my opinion.

Reply:
Mine would be:
```
setOpen(state => {
  const newVal = !state;
  trackOpenClose(newVal);
  return newVal;
})
```